acorn/src/regexp.js

```diff
@@ -920,6 +984,238 @@ pp.regexp_eatClassEscape = function(state) {
   )
 }
 
+// https://tc39.es/ecma262/#prod-ClassSetExpression
+// https://tc39.es/ecma262/#prod-ClassUnion
+// https://tc39.es/ecma262/#prod-ClassIntersection
+// https://tc39.es/ecma262/#prod-ClassSubtraction
+pp.regexp_classSetExpression = function(state) {
+  let nextMayContainStrings = false
+  if (this.regexp_eatClassSetRange(state)) {
+    // Continue with ClassUnion processing.
+  } else if (this.regexp_eatClassSetOperand(state)) {
+    nextMayContainStrings = state.lastMayContainStrings
+    let pos = state.pos
+    const start = pos
+    // https://tc39.es/ecma262/#prod-ClassIntersection
+    while (
+      state.eatChars([0x26, 0x26] /* && */) &&
+      state.current() !== 0x26 /* & */ &&
+      this.regexp_eatClassSetOperand(state)
+    ) {
+      if (!state.lastMayContainStrings) nextMayContainStrings = false
+      pos = state.pos
+    }
+    state.pos = pos
+    if (state.pos !== start) {
+      state.lastMayContainStrings = nextMayContainStrings
+      return
+    }
+    // https://tc39.es/ecma262/#prod-ClassSubtraction
+    while (
+      state.eatChars([0x2D, 0x2D] /* -- */) &&
+      this.regexp_eatClassSetOperand(state)
+    ) {
+      pos = state.pos
+    }
+    state.pos = pos
```

Comment:
Is it actually valid for `--`/`&&` to occur but then not be followed by a set operand? If not, why not move the parsing of the operand into the `if` and make it raise rather than return a boolean?

Reply:
I changed it to raise an error if the operator is not followed by an operand.
https://github.com/acornjs/acorn/pull/1219/commits/0265d1b5933e49841c50088b956abd3760155097